### PR TITLE
[DLK-200] Add documentation about data retention policies - Merge 7/23

### PR DESCRIPTION
**Pull request by asimonutti**
### Summary

Added documentation to the Snowflake Data Sharing docs about our data retention policies, and what happens when we archive event data.

### Related PRs, issues, or features (optional)

- [DLK-191](https://jira.braze.com/browse/DLK-191)

### Feature release date (optional)

- August 20th 2024

### Contributor checklist

- [ ] I confirm that my PR meets the following:
    - I signed the [Contribution License Agreement (CLA)](https://www.braze.com/docs/cla).
    - My style and voice follows the [Braze Style Guide](https://docs.google.com/document/u/2/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub).
    - My content contains correct spelling and grammar.
    - All links are working correctly.
    - If I renamed or moved a file or directory, I set up [URL redirects](https://www.braze.com/docs/contributing/content_management/redirecting_urls) for each file.
    - If I updated or replaced an image, I did not remove the original image file from the repository. (For more information, see [Updating an image](https://www.braze.com/docs/contributing/content_management/images/#updating-an-image)).
    - If my PR is related to a paid SKU, third party, SMS, AI, or privacy, I have received written approval from Braze Legal.

### Submitting for review

If your PR meets the above requirements, select **Ready for review**, then add a reviewer:
- **Non-Braze contributors:** Add `braze-inc/docs-team` as the reviewer.
- **Braze employees:** Have any relevant subject matter experts (like engineers or product managers) review your work first, then add the [tech writer assigned to your specific vertical](https://confluence.braze.com/pages/viewpage.action?pageId=75039896). If you're not sure which tech writer to add, you can add `braze-inc/docs-team` instead. 

Thanks for contributing! We look forward to reading your work.

### CODE REVIEW

**Review by colleenkerr — APPROVED** [review]

**Review by julianw-braze — APPROVED** [review]

File: _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md [hunk 1]
```diff
@@ -103,7 +103,14 @@ When possible, breaking changes will be preceded by an announcement and a migrat
...
+
+#### Retention Policy
+Braze will store all event data for up to 2 years. Any data older than 2 years will be archived and moved to cold storage. As part of the archival process all events will be anonymized and any PII sensitive fields will be stripped out (this includes optionally PII fields like `properties`), however this data will still contain the `user_id` field which will remain consistent and can be tracked across both hot and cold tables.
+
+You will be able to query against the most recent 2 years of data for each event in the corresponding `USERS_*_SHARED` view. Additionally, each event will have a `USERS_*_SHARED_ALL` view which can be queried against to return all data across both hot and cold storage tables.
+
+#### Historical data
 Braze's historical event data in Snowflake goes back to April 2019. In the first few months of Braze storing data there, product changes were made that may have resulted in some of that data looking slightly different or having some null values (as we weren't passing data into every available field at this time). It's best to assume that any results that include data before August 2019 may look slightly different from expectations.
```
Comment (fanley) [inline _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md:114]: Does this need to be updated as part of the update?

Comment (asimonutti) [reply _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md:114]: This is still true even after the move to `_ARCHIVED` tables.

Comment (fanley) [reply _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md:114]: `Braze's historical event data in Snowflake goes back to April 2019`

and

`Braze will store all event data for up to 2 years`

read as conflicting statements. Am I missing something?

Comment (asimonutti) [reply _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md:114]: I guess the `all` is the important bit. We will continue to store historical data back to 2019, just with the PII fields nulled out.

Comment (asimonutti) [reply _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md:114]: @fanley do you have any suggestions for how I could reword this to be more clear?

Comment (josh-mccrowell-braze) [reply _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md:114]: Proposing this rewrite: "The archive of historical event data in Snowflake goes back to April 2019."

This helps bridge the apparent contradiction between how long Braze keeps the data before archiving it, and avoids the construction "Braze's", which our style guide prefers us not to use.

File: _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md [hunk 2]
```diff
@@ -103,8 +103,15 @@ When possible, breaking changes will be preceded by an announcement and a migrat
 ### Snowflake regions
 Braze currently hosts all user-level data in the Snowflake AWS US East-1 and EU-Central (Frankfurt) regions. For users outside of those regions, Braze can provide data sharing to joint customers who are hosting their Snowflake infrastructure across any AWS, Azure, or GCP region.
 
-### Historical data
-Braze's historical event data in Snowflake goes back to April 2019. In the first few months of Braze storing data there, product changes were made that may have resulted in some of that data looking slightly different or having some null values (as we weren't passing data into every available field at this time). It's best to assume that any results that include data before August 2019 may look slightly different from expectations.
+### Data Retention
+
+#### Retention Policy
+Any data older than two years will be archived and moved to longer term storage. As part of the archival process, all events are anonymized and any personal identifiable information (PII) sensitive fields is stripped out (this includes optionally PII fields like `properties`). Archived data still contains the `user_id` field, which allows for per-user analytics across all events data.
```
Comment (josh-mccrowell-braze) [inline _docs/_partners/data_and_infrastructure_agility/data_warehouses/snowflake.md:109]:
```suggestion
Any data older than two years will be archived and moved to long term storage. As part of the archival process, all events are anonymized and any personal identifiable information (PII) sensitive fields are stripped out (this includes optionally PII fields like `properties`). Archived data still contains the `user_id` field, which allows for per-user analytics across all events data.
```

**Review by josh-mccrowell-braze — APPROVED** [review]
